odpc.client/src/features/publicatie/composables/use-paged-search.ts

```diff
@@ -0,0 +1,123 @@
+import { ref, watch, computed, onMounted, type Ref } from "vue";
+import { useRouter } from "vue-router";
+import { useUrlSearchParams, type UrlParams } from "@vueuse/core";
+import { useFetchApi } from "@/api/use-fetch-api";
+
+const API_URL = `/api/v1`;
+const PAGE_SIZE = 5;
+
+export type PagedResult<T> = {
+  count: number;
+  next?: string;
+  previous?: string;
+  results: T[];
+};
+
+export const usePagedSearch = <T, U extends string>(endpoint: string, params: UrlParams) => {
```

Comment:
dus je instantieert een anoniem object in een types.ts, om daar een PublicatieSearchParam type uit te extraheren, om die hier te gebruiken om via een Record tot een QueryParam type te komen, om .... 🥵 pfff het mag allemaal heeel veel minder abstract en generiek. liever gewoon iets leesbaars en begrijpelijks dat doet wat het hier moet doen en ook onderhouden kan worden door mere human beings .

Reply:
Ik heb verwijzingen naar types weggehaald, was na enige refactoring idd onduidelijke plek. Blijft over een volgens mij vrij standaard composable patroon met config objectje. Nog wat code opgeschoond en comments toegevoegd.

Heb naar generieke oplossing gestreefd om op later moment eenvoudig meer filters te kunnen toevoegen (configureren), bijvoorbeeld voor status, waardelijsten etc. En deze generieke opzet kan denk ik straks gebruikt worden voor ODBP. Composable uit feature gehaald.